File: LeetCode_Solutions/LeetCode_Solutions/InvertBinaryTree_226.m
//
//  InvertBinaryTree.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/10/9.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//


#import "InvertBinaryTree_226.h"

struct TreeNode {
    int val;
    struct TreeNode *left;
    struct TreeNode *right;
};

@implementation InvertBinaryTree_226

- (struct TreeNode *)invertTree:(struct TreeNode *)root {

    if (!root) {
        return nil;
    }
    struct TreeNode *temp = root->left;
    root->left = root->right;
    root->right = temp;
    
    [self invertTree:root->left];
    [self invertTree:root->right];
    return root;

}

@end

File: LeetCode_Swift/LeetCode_Swift/InvertBinaryTree_226.swift
//
//  InvertBinaryTree_226.swift
//  LeetCode_Swift
//
//  Created by Huni on 07/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation

public class InvertBinaryTree_226 {

    func invertTree(_ root: TreeNode?) -> TreeNode? {

        guard root != nil else {
            return root
        }
        
        let tempTree = root?.left
        root?.left = root?.right
        root?.right = tempTree
        
        _ = invertTree(root?.left)
        _ = invertTree(root?.right)
        
        return root
        
    }

}
